fdbserver/BlobWorker.actor.cpp

```diff
@@ -155,6 +156,16 @@ struct GranuleHistoryEntry : NonCopyable, ReferenceCounted<GranuleHistoryEntry>
 	  : range(range), granuleID(granuleID), startVersion(startVersion), endVersion(endVersion) {}
 };
 
+// TODO: does this need to be versioned like SS has?
```

Comment:
The SS versions the tenant map so that we can only access tenants at version where they exist. If we delete a tenant at version X, for example, we cannot read or write keys with it in X+1.

I'm not sure if reading stale tenants would cause any problems (I suppose it could by violating causal consistency), but writing them seems more problematic.

Reply:
Yeah. My logic here was that, since it is versioned here, and the blob worker only picks up changes from the SS, the SS knowing that it is versioned mostly removes the need for the blob manager and worker to have it versioned.

I also explicitly punted on doing anything about deleted tenants in this PR. We will need a better solution eventually, but right now these simulation tests don't test that. I think it may be the case that with deleting tenants, we need it to be versioned still?

The manager only cares about the tenant boundaries for granule range boundaries and what to blobbify, so i don't think it needs to be versioned.
For reading a new tenant, if a blob worker read races with a tenant creation that the worker doesn't yet know about, it just tells the client to retry. This also works for now because it doesn't handle deletes.
This will need a better solution in the future though, and I have some ideas.

Reply:
Sounds good, I think if we are ignoring deletions for now then versioning shouldn't be necessary.